explorer/interpreter/type_checker.cpp

```diff
@@ -3075,17 +3075,38 @@ class AbstractPattern {
 // the algorithm used here.
 class PatternMatrix {
  public:
+  // Add a pattern vector row to this collection of pattern vectors.
+  void Add(std::vector<AbstractPattern> pattern_vector) {
+    CARBON_CHECK(matrix_.empty() || matrix_[0].size() == pattern_vector.size());
+    matrix_.push_back(std::move(pattern_vector));
+  }
+
+  // Is the given pattern vector redundant if it appears after the patterns in
+  // this matrix? That is, will it never match following the other patterns?
```

Comment:
```suggestion
  // Returns true if the given pattern vector is redundant when it appears after the patterns in
  // this matrix. That is, will it never match following the other patterns?
```

Comments should clearly describe behavior to the reader; only asking questions may leave readers wondering who's supposed to answer.
https://google.github.io/styleguide/cppguide.html#Function_Comments